src/daemon/settlement.rs

```diff
@@ -0,0 +1,182 @@
+use std::{
+    fmt,
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use anyhow::Context;
+use blake3::Hash;
+use bytes::Bytes;
+use dashmap::DashMap;
+use earendil_crypt::{Fingerprint, IdentityPublic, IdentitySecret};
+use serde::{Deserialize, Serialize};
+use smol::channel::{Receiver, Sender};
+use stdcode::StdcodeSerializeExt;
+
+use super::context::{DaemonContext, DEBTS, GLOBAL_IDENTITY};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SettlementRequest {
+    // unix milliseconds
+    timestamp_ms: u64,
+    pub decrease: u64,
```

Comment:
this does not need to be `pub`? The only place that this field is used is in the same module.

Note that public/private visibility in Rust is at the *module* level, not at the type level, unlike OOP languages. So it's perfectly fine for a method of Settlements to call something private in SettlementRequest as long as they're in the same module.